datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -264,9 +264,9 @@ impl<F: FileOpener> FileStream<F> {
                     Ok(reader) => {
                         let partition_values = mem::take(partition_values);
 
-                        let next = self.start_next_file().transpose();
-
+                        // include time needed to start opening in `start_next_file`
```

Comment:
With this change, the FileStream now consistently includes the time for `start_next_file` in `time_opening` -- both here and immediately above:


https://github.com/apache/arrow-datafusion/blob/5c0fe0d7ebe0b205ce10c2b07ee2230ff3df6cba/datafusion/core/src/physical_plan/file_format/file_stream.rs#L244-L246

Reply:
I think the `self.start_next_file().transpose()` return a `Future` and the call should take almost no time.  
Sorry that I do not get why making this change.

Reply:
> return a Future and the call should take almost no time.
> Sorry that I do not get why making this change.

It was suggested by @thinkharderdev  in https://github.com/apache/arrow-datafusion/pull/5790#discussion_r1155734134: 

> I think we want this call before self.start_next_file. In some cases (for example, when reading from the LocalFileSystem store outside of a Tokio context) this can block and we would not capture the time spent on disk IO.